docs/contributing/add_new_resource_tgc_next.md

```diff
@@ -0,0 +1,330 @@
+
+	It might require to install relevant go version for this to run successfully. Follow the [Before you begin](https://googlecloudplatform.github.io/magic-modules/get-started/generate-providers/#before-you-begin) and [Setup your development environment](https://googlecloudplatform.github.io/magic-modules/develop/set-up-dev-environment/) sections from the [Magic Modules documentation](https://googlecloudplatform.github.io/magic-modules/get-started/generate-providers/).
+
+##### Run unit tests
+To run the unit tests locally, run the following from the root of the `terraform-google-conversion` repository:
+
+```
+make test
+```
+
+##### Run integration tests
+To run the integration tests for the added resource locally, run the following from the root of the `terraform-google-conversion` repository:
+```
+make test-integration TESTPATH=./test/services/alloydb  TESTARGS='-run=TestAccAlloydbBackup_alloydbBackupBasicTestExample' > alloydbBackup.log
+```
+
+To run one integration test for the added resource locally, run the following from the root of the `terraform-google-conversion` repository:
+```
+make test-integration TESTPATH=./test/services/alloydb  TESTARGS='-run=TestAccAlloydbBackup_alloydbBackupBasicTestExample' > alloydbBackup.log
+```
+
+The core integration tests in `terraform-google-conversion` mirror the naming of the corresponding acceptance tests in the Terraform provider. This testing process uses a crucial round-trip validation method:
+
+1. Test Input Data
+
+	The input for each equivalent integration test combines two elements:
+
+	* The original Terraform resource configuration from the acceptance test step.
+
+	* The Cloud Asset Inventory (CAI) data exported from the actual resources provisioned during the nightly execution of the Terraform provider tests.
+
+2. Test File Generation (WRITE_FILES Flow)
+
+	By setting the environment variable WRITE_FILES, running the integration tests locally generates a series of files in the service folder, detailing the entire conversion cycle.
+
+	For example, running `TestAccAlloydbBackup_alloydbBackupBasicTestExample` successfully creates five files for each step:
+
+	| File | Content | Conversion Step |
+	| :--- | :--- | :--- |
+	| File 1 (.tf) | Original Terraform configuration (raw_config) | Start |
+	| File 2 (.json) | CAI Assets exported from the resource (export_assets) | Input for CAI to HCL |
+	| File 3 (export.tf) | Converted Terraform configuration (export_config) from File 2 | cai2hcl |
+	| File 4 (roundtrip.json) | CAI Assets (roundtrip_assets) converted from File 3 | tfplan2cai |
+	| File 5 (roundtrip.tf) | Final Terraform configuration (roundtrip_config) from File 4 | cai2hcl |
+
+	The integration tests pass only when two conditions are met:
+	1. Every field in the original `raw_config` must exist within the generated `export_config`.
+	2. The `export_config` and the final `roundtrip_config` must be identical.
+
+##### Address integrations tests failures
```

Comment:
It would be nice to have some examples of error messages for this section so that searches for error messages can find the relevant section.